test/include/dlaf_test/eigensolver/test_eigensolver_correctness.h

```diff
@@ -58,6 +58,11 @@ void testEigensolverCorrectness(const blas::Uplo uplo, Matrix<const T, Device::C
     return allGather(blas::Uplo::General, mat_e.get(), grid...);
   }();
 
+  // eigenvalues are contiguous in the mat_local buffer
+  BaseType<T>* evals_start = mat_evalues_local.ptr({0, 0});
+  BaseType<T>* evals_end = evals_start + m;
+  EXPECT_TRUE(std::is_sorted(evals_start, evals_end));
```

Comment:
This makes me wonder if it would make sense to provide iterators/begin/end at least for some of our matrix/tile types (obviously only where it makes sense, layout concerns may make this harder than it seems)? Definitely not for this PR, but worth at least tracking in a low priority issue?

Reply:
This works because we are dealing with a `MatrixLocal` (used only in tests) used as a column vector.

Reply:
It's clear to me why this works. My comment was about allowing to write `std::is_sorted(std::begin(mat_evalues_local), std::end(mat_evalues_local))`. But perhaps you're saying that the number of cases where this will work are too few so it's not worth trying to provide that functionality?